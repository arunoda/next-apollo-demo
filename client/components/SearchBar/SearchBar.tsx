import React from 'react';
import { FunctionComponent } from "react"

import styles from './searchbar.module.scss';

interface Props {
  setInputValue: (inputValue: string) => void
}

export const SearchBar: FunctionComponent<Props> = ({ setInputValue }) => {

  const onChange = (e: any) => {
    const value = e.target.value
    if (value.length >= 3)
      setInputValue(value)
    else setInputValue('')
  }

  return (
    <>
      <input
        className={styles.input}
        name='searchBar'
        type='text'
        onChange={onChange}
        data-testid='searchBar'
        placeholder='Search...'
      />
    </>
  )
}