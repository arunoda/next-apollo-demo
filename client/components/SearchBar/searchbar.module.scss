.input {
  width: 100%;
  height: 30px;
  border-radius: 15px;
  padding: 0 8px;
  box-sizing: border-box;
}
