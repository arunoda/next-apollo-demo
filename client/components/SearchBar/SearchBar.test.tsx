import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import { SearchBar } from "./SearchBar";

describe('SearchBar', () => {
  const setInputValue = jest.fn();

  it('should match snapshot', () => {
    const { container } = render(<SearchBar setInputValue={setInputValue} />);
    expect(container).toMatchSnapshot();
  });

  it('should trigger the props function when typing the input', () => {
    const { getByTestId } = render(<SearchBar setInputValue={setInputValue} />);
    fireEvent.change(getByTestId('searchBar'), { target: { value: 'miss' } });
    expect(setInputValue).toHaveBeenCalledTimes(1);
  })
})