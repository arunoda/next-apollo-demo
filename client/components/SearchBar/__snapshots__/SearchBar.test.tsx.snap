// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<div>
  <input
    class="input"
    data-testid="searchBar"
    name="searchBar"
    placeholder="Search..."
    type="text"
  />
</div>
`;
